torch/_export/serde/serialize.py

```diff
@@ -2791,3 +2806,37 @@ def replace_output(out):
         schema_version=ep.schema_version,
         dialect=ep.dialect
     )
+
+
+class CustomOpHandler:
+    """
+    Base class for handling custom operators.
+    """
+    def namespace(cls):
+        raise NotImplementedError(f"{cls.__class__} namespace() must be implemented")
+
+    def op_name(cls, name):
+        raise NotImplementedError(f"{cls.__class__} op_name() must be implemented")
+
+
+def register_custom_op_serialization(
+    op_handler: CustomOpHandler,
+    serialization_fn: Callable[[Tuple[GraphModuleSerializer, torch.fx.Node], Node], Node],
+):
+    """Register custom serialization method for a node."""
+    assert isinstance(op_handler, CustomOpHandler), f"Expected CustomOpHandler, got {type(op_handler)}."
+
+    namespace = op_handler.namespace()
+    # FIXME: hardcoded target because currently custom op serialization only handles
+    # node.target during serialization.
+    op_name = op_handler.op_name("target")
+
+    if namespace not in _serialization_registry:
+        _serialization_registry[namespace] = {}
+    _serialization_registry[namespace][op_name] = serialization_fn
+
```

Comment:
We should create two tables, one for serialization and the other for deserialization.
```suggestion
    _serialization_registry[op_type] = op_handler
```